tests/docker_cluster.py

```diff
@@ -319,10 +355,12 @@ def clean_up_presto_test_images(self):
 
     def get_ip_address_dict(self):
         ip_addresses = {}
-        all_hosts = self.all_hosts()
-        for host in all_hosts:
+        for host, internal_host in zip(self.all_hosts(),
+                                       self.all_internal_hosts()):
             inspect = self.client.inspect_container(host)
             ip_addresses[host] = inspect['NetworkSettings']['IPAddress']
+            ip_addresses[internal_host] = \
```

Comment:
I don't think so. What you end up here is a dict whose keys are the set of all unique hosts + the set of all internal hosts. That way if you query the dict with a unique or internal host, you'll get the right IP. The iteration happens with a tuple: master (being the internal host) and master-alksdjf-afsdgas-asdf (being the unique host).